src/components/specific/models/models-card/main-model-selector/MainModelSelector.vue

```diff
@@ -0,0 +1,111 @@
+<script setup>
+import { ref, watch } from "vue";
+import { MODEL_TYPE } from "../../../../../config/models.js";
+import { useModels } from "../../../../../state/models.js";
+import { useProjects } from "../../../../../state/projects.js";
+
+const { IFC, POINT_CLOUD } = MODEL_TYPE;
+const { loadProjectModels } = useModels();
+const { updateProject } = useProjects();
+
+const props = defineProps({
+  project: {
+    type: Object,
+    required: true
+  },
+  models: {
+    type: Array,
+    required: true
+  },
+  onCancel: {
+    type: Function
+  },
+  onSuccess: {
+    type: Function
+  },
+  onError: {
+    type: Function
+  }
+});
+
+const loading = ref(false);
+const selectedModel = ref(null);
+
+watch(
+  () => props.models,
+  () => {
+    selectedModel.value =
+      props.models.find(m => m.id === props.project.main_model_id) ??
+      props.models[0];
+  },
+  { immediate: true }
+);
+
+const cancel = () => {
+  props.onCancel?.();
```

Comment:
Are there cases where onCancel in not defined?

I prefer non-nullable props than `?.` everywhere when possible

Reply:
or a default props to `() => {}`

Reply:
I think I prefer `?.` than `() => {}` defined somewhere else.

But I don't why we would need nullable fields on a component used in one place